base/util.go

```diff
@@ -1307,10 +1307,11 @@ func InjectJSONProperties(b []byte, kvPairs ...KVPair) (new []byte, err error) {
 			valBytes = []byte(strconv.FormatUint(uint64(v), 10))
 		case uint64:
 			valBytes = []byte(strconv.FormatUint(v, 10))
-		case string:
-			valBytes = []byte(`"` + v + `"`)
 		case bool:
 			valBytes = []byte(strconv.FormatBool(v))
+		case string:
+			// it's not safe to use strings without marshalling
+			valBytes = []byte(ConvertToJSONString(v))
```

Comment:
I can't think of a circumstance where this would happen where `ConvertToJSONString` would fail in practice unless there were non utf-8 strings. In this case, I don't think this function would be called because it should only be called on non binary (already json according to CBS) documents.

To be pedantic, you could include string in the default case because then `JSONMarshal` error value is checked.

Reply:
Updated, just falling through to `default` implicitly with a comment why.